### PR TITLE
Implemented test_top_similar_traces_CLI.py

### PR DESCRIPTION
**Pull request by aditya-iitd**
A CLI interface that does something similar to what "test_top_similar_traces.py" does:
 - the user gives in input a stack trace;
 - the CLI tool downloads some data (e.g. the past 7 days), trains the model on that data and returns the top ten similar stack traces.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 .DS_Store
 *.json
 crashclustering_data/schema_version
+
```
Comment (marco-c) [inline .gitignore]: Nit: can you remove this blank line?

File: crash_similarity.py [hunk 2]
```diff
@@ -85,7 +85,8 @@ def get_stack_trace_for_uuid(uuid):
     return data['proto_signature']
 
 
-def train_model(corpus):
+ def train_model(corpus):
+
```
Comment (marco-c) [inline crash_similarity.py]: Nit: can you remove the blank line and the space before def?

File: test_top_similar_traces_CLI.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+
```
Comment (marco-c) [inline test_top_similar_traces_CLI.py]: Nit: can you remove the initial blank line?

File: test_top_similar_traces_CLI.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# CLI INTERFACE THAT TAKES STACK TRACE AS INPUT AND RETURNS SIMILAR STACK TRACES
+
+import download_data
+import crash_similarity
+import argparse
+import utils
+from datetime import timedelta
```
Comment (marco-c) [inline test_top_similar_traces_CLI.py]: Nit: `utils` and `timedelta` are unused, can you avoid importing them?

File: crash_similarity.py [hunk 5]
```diff
@@ -84,7 +84,6 @@ def get_stack_trace_for_uuid(uuid):
     data = download_data.download_crash(uuid)
     return data['proto_signature']
 
-
```
Comment (marco-c) [inline crash_similarity.py]: Nit: can you avoid removing the blank line here?

**Comment by marco-c** [comment]
There are some issues in the tests run: https://travis-ci.org/marco-c/crashsimilarity/jobs/212187647#L216.

**Comment by aditya-iitd** [comment]
I don't know why i am getting ' EOF in multi-line statement ' error!

File: test_top_similar_traces_CLI.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import download_data
+import crash_similarity
+import argparse
+
+parser = argparse.ArgumentParser(description='Returns the top ten similar stack traces')
+parser.add_argument('--crash_id', required=True, help='crash_id corresponding to the stack trace')
+parser.add_argument('--product', required=True, help='Product for which crash data is needed to be downloaded')
+parser.add_argument('--top', help='Number of top similar and different stack traces(Default 10)', default=10, type=int)
+args = parser.parse_args()
+if __name__ == '__main__':
```
Comment (marco-c) [inline test_top_similar_traces_CLI.py]: You removed all white spaces 😄 
Can you put some back, as it was before?

File: test_top_similar_traces_CLI.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    download_data.download_crashes(days=7, product=args.product)
+    paths = download_data.get_paths(days=7, product=args.product)
+    corpus = crash_similarity.read_corpus(paths)
+    model = crash_similarity.train_model(corpus)
+    stack_trace = crash_similarity.get_stack_trace_from_crashid(args.crash_id)
+    similarities = crash_similarity.top_similar_traces(model, corpus, stack_trace, args.top)
+
+    for similarity in similarities:
+        print(u'%s: <%s>\n' % ((corpus[similarity[0]].tags[1], similarity[1]), ' '.join(corpus[similarity[0]].words))
```
Comment (marco-c) [inline test_top_similar_traces_CLI.py]: There's a missing parenthesis at the end of this line

Comment (marco-c) [inline test_top_similar_traces_CLI.py]: Ahah, I know the feeling 😄

**Comment by marco-c** [comment]
Thanks!

**Pull request merged**